### PR TITLE
Add few docs and improvement in check_ip_address

### PR DESCRIPTION
**Pull request by jkonecny12**
``check_ip_address()`` is more readable and has documentation now.
Some functions get new documentation strings.

### CODE REVIEW

File: pyanaconda/network.py [hunk 1]
```diff
@@ -116,8 +126,11 @@ def sanityCheckHostname(hostname):
 
     return (True, "")
 
-# Return a list of IP addresses for all active devices.
 def getIPs():
+    """ Return a list of IP addresses for all active devices.
+
+        :return: list of IP addresses
```
Comment (vpodzime) [inline pyanaconda/network.py]: I think this only duplicates the description. Either drop it and make the docstring one-line or replace it with ``:rtype:`` which would be more interesting here, I'd say.

File: pyanaconda/network.py [hunk 2]
```diff
@@ -130,14 +143,19 @@ def getIPs():
     # prefer IPv4 addresses to IPv6 addresses
     return ipv4_addresses + ipv6_addresses
 
-# Return the first real non-local IP we find
 def getFirstRealIP():
+    """ Return the first real non-local IP we find from the list of
+        all active devices.
+
+        :return: first non-local IP or ``None``
```
Comment (vpodzime) [inline pyanaconda/network.py]: Same here.

File: pyanaconda/network.py [hunk 3]
```diff
@@ -161,9 +179,12 @@ def prefix2netmask(prefix):
     netmask = ".".join(str(byte) for byte in _bytes)
     return netmask
 
-# Try to determine what the hostname should be for this system
 def getHostname():
+    """ Try to determine what the hostname should be for this system
 
+        :return: Hostname for this system
```
Comment (vpodzime) [inline pyanaconda/network.py]: this can also be dropped

**Comment by vpodzime** [comment]
Looks good to me otherwise.

File: pyanaconda/network.py [hunk 4]
```diff
@@ -80,6 +80,16 @@ def setup_ifcfg_log():
     ifcfglog = logging.getLogger("ifcfg")
 
 def check_ip_address(address, version=None):
+    """
+    Check if the given IP address is valid in given version if set.
+
+    :param str address: IP address for testing
+    :param int version: ``4`` for IPv4, ``6`` for IPv6 or
+                        nothing if we don't care
```
Comment (bcl) [inline pyanaconda/network.py]: I think it would be clearer to say something like "or None to allow either format."

**Comment by jkonecny12** [comment]
@vpodzime , @bcl thank you for responses. I did the changes and updated this PR.

**Comment by vpodzime** [comment]
Looks good to me now.

**Comment by jkonecny12** [comment]
Pushed

**Pull request closed without merging**